charmstore_internal_test.go

```diff
@@ -177,12 +177,13 @@ func (ResourceSuite) TestPublish(c *gc.C) {
 		"data": 1,
 		"lib":  2,
 	}
-	result, err := s.Publish(id, resources)
+	result, err := s.Publish(id, []string{"development"}, resources)
```

Comment:
You should probably test the "no channels == stable" thing. Or just delete that logic, which I think I prefer.